src/entry_points/gentest.py

```diff
@@ -0,0 +1,303 @@
+"""
+Define an entry point wrapper for test generator.
+"""
+
+import json
+import os
+import sys
+from dataclasses import asdict, dataclass
+from typing import Dict, List, Union
+
+import requests
+
+from ethereum_test_tools import Account, Address, Transaction
+
+
+def main():  # noqa: D103
+    if len(sys.argv) != 3:
+        print_help()
+        sys.exit(1)
+    if sys.argv[1] in ["-h", "--help"]:
+        print_help()
+        sys.exit(0)
+    make_test(sys.argv)
+
+
+def print_help():  # noqa: D103
+    print("Extracts transaction and required state from the mainnet to make a BC test out of it")
+    print("Usage: gentest <tx_hash> <file.py>")
+
+
+if __name__ == "__main__":
+    main()
+
+
+def make_test(args):  # noqa: D103
+    transaction_hash = args[1]
+    output_file = args[2]
```

Comment:
We normally use `argparse` to parse the arguments for an entry point: https://github.com/ethereum/execution-spec-tests/blob/06ed07f6ec5e8bbceb0f05443ba6ddfe9c85613c/src/entry_points/hasher.py#L113-L120

See an tutorial here: https://docs.python.org/3/library/argparse.html